src/test/test_dir.c

```diff
@@ -1566,18 +1566,55 @@ test_dir_measured_bw_kb(void *arg)
 static void
 test_dir_dirserv_read_measured_bandwidths(void *arg)
 {
-  char *fname=NULL;
   (void)arg;
+  char *content = NULL;
+  time_t timestamp = time(NULL);
+  char *fname = tor_strdup(get_fname("V3BandwidthsFile"));
+  smartlist_t *bwlist_headers = smartlist_new();
+  char *bwlist_headers_str = NULL;
+  char *out_bwlist_headers_str = NULL;
 
-  fname = tor_strdup(get_fname("V3BandwidthsFile"));
   /* Test an empty file */
   write_str_to_file(fname, "", 0);
   setup_capture_of_logs(LOG_WARN);
-  tt_int_op(-1, OP_EQ, dirserv_read_measured_bandwidths(fname, NULL));
+  tt_int_op(-1, OP_EQ, dirserv_read_measured_bandwidths(fname, NULL, NULL));
   expect_log_msg("Empty bandwidth file\n");
 
+  /* Test v1.1.0 headers */
```

Comment:
Please add another test for v1.0.0 headers. (That is, just a timestamp.)
Please add a test for v1.1.0 headers with no terminator (it is a SHOULD in the spec).
Please add a test for all 3 kinds of headers (v100, v110-terminator, v110-no-terminator) with relay lines.

You can also add tests with malformed relay lines if you want: any malformed lines before the terminator (if present) or the first valid relay line will end up being treated as headers.

Reply:
I think i should move the 1st comparison of this test (bandwidth file empty) to https://gitweb.torproject.org/tor.git/tree/src/test/test_dir.c#n1649, that has been recently merged, and the name of this test to test_dir_dirserv_read_measured_bandwidths_bwlist_headers, since what we want to check here is the generation of the bwlist_headers.

Reply:
Sounds like a good idea to me!